chore: re-write release docs (#2137)
projen/projen: docs/releases.md

releaseTrigger: ReleaseTrigger.scheduled({ schedule: '0 17 * * *' }),
```

> ## Local Releases

Reviewer: I'm not sure, I feel like "manual" is still a better term here.

Author: Yeah, I'm also not 100% sure. My feeling was that the term "manual" could mean a lot of different things. I'll sleep over it.

Author: Going with manual release again. My reason was mostly to pre-empt a hypothetical future feature to trigger workflow based releases only on request. We can name that differently though.

If you don't want projen to automatically release your project, you can configure a manual release trigger:

```js
releaseTrigger: ReleaseTrigger.manual(),
```

> This will create a `release` task. Run it locally to cut a release. It will build the project and create releasable artifacts inside the `dist` directory.

Reviewer: ```suggestion
This will create a `release` task. Run it locally (`npx projen release`) to cut a release. It will build the project and create releasable artifacts inside the `dist` directory.
```

> This will create a `release` task. Run it locally (`npx projen release`) to cut a release. It will build the project and create releasable artifacts inside the `dist` directory.

Author: ```suggestion
This will create a `release` task. Run it locally (`projen release`) to cut a release. It will build the project and create releasable artifacts inside the `dist` directory.
```

Author: I believe the convention in the docs is to not include `npx`

It will also trigger a `publish:git` task. This task does
manage a project-level changelog, commit any changes, tag the release, and push any commits and tags to the remote repository.

The command used for pushing can be customized by specifying
`gitPushCommand`. Set to an empty string to disable pushing entirely.

### Publishing modules for local releases

> Publishing modules to their respective package repositories is modelled as a separate task in Projen. While automated release triggers take care of this for you, this is **NOT** the case when using `ReleaseTrigger.manual()`.

Reviewer: The "separate task" throws me off, especially since if they are there, they don't seem to be used in the examples below. The most important part seems to be whether publishing is automatic or not. Maybe something like:

```suggestion
If you are using manual releases (`ReleaseTrigger.manual()`), Projen will *not* automatically publish packages to their respective package repositories. 
You are expected to publish...
```

> You are expected to publish the release artifacts from the `dist` directory manually. **Do not attempt to publish the repository directly, it will fail.**

Reviewer: I don't know what it means to publish a repository directly. Should I?

Author: @rix0rrr 
It refers to the common practice in npm land (and presumingly other package manager ecosystems) to run `npm publish` at the root of the repository which will then take care of packaging and uploading for the user. It's also the first thing users attempt after they decide not to use the provided publish workflow.

Maybe `to publish from the repository directly` is bit clearer or maybe it's not needed at all?

For example in a Node.js project, you might run:

> - `projen release`

Reviewer: Does this also do a build?

Author: Yes

- `npm publish dist/js/my-package-1.2.3.tgz`

Or for a multi language jsii project, the necessary steps could look something like:

- `projen release`
- `npx -p publib@latest publib-npm`
- `npx -p publib@latest publib-pypi`
- `npx -p publib@latest publib-nuget`

It is also your responsibility to ensure credentials are setup and available for each package repository published to.

## FAQ

### How can I force a different version?

You may be adopting Projen in a project that has already been published to previous versions. Projen uses tags to determine the version to start with
before bumping according to the rules previously mentioned.

You can force the base version number by adding a tag to your repo. For example, if the latest version of your project is 1.2.3, then add a tag to your main branch of `v1.2.3`.
The next time the release workflow runs, it will bump from version 1.2.3.

> As explained above, this is based on the conventional commits so  the next release will either be 1.2.4 or 1.3.0.

Reviewer 2: ```suggestion
As explained above, this is based on the conventional commits so the next release will either be 1.2.4, 1.3.0 or 2.0.0 on a breaking change.
```
